### DISCUSSION TITLE
Check if any answer was ever given

### DISCUSSION THREAD

**Discussion by Marcelo-Petrucelli** [orestbida/cookieconsent#500] [Q&A] [answered]
Hello guys. Thanks for this amazing project.

Is there any way of knowing if any interaction with the consent was done?

I mean, my problem is that I wait the user's answer to unpause my google ads from loading. That's fine because "onAccept" runs with the Run method. But this only happens if, and only if, the user has given any answer to the consent.

If the user has already interacted with the consent banner the page loads fine, because onAccept runs and I'm able to see if the ad category is not selected and work over my ads to make them non-personalized. On the other hand if they are, I simply unpause my ads.

But if the user simply don't answer the banner at all, or even don't interact with the consent, I'm not able to know if I should unpause my ad loading. I could set up some type of timeout but that's definately a horrible choice.

Any suggestions or ways to find out if I'm in the "initial state" of the consent?

--- Extra ---

I also noticed that, even when my config is set to Opt-out, when the user has not yet interacted with my consent banner I get from a ".getUserPreferences()"'s call:
```
accept_type: "custom"
accepted_categories: [ ]
rejected_categories: ['necessary', 'analytics', 'ads', 'social']
```

**Comment by orestbida** [comment, marked as answer]
Hi @Marcelo-Petrucelli,

you can "force" the user to make a choice before browsing the website via [`force_consent: true`](https://github.com/orestbida/cookieconsent#all-configuration-options). 

> Any suggestions or ways to find out if I'm in the "initial state" of the consent?

You can check if the plugin's cookie is set or not.

> I also noticed that, even when my config is set to Opt-out, when the user has not yet interacted with my consent banner ...

That's the expected behavior, `getUserPreferences()` only reflects the user's preferences (if expressed).
